switchlink/switchlink_main.c

```diff
@@ -294,6 +305,13 @@ struct nl_sock *switchlink_get_nl_sock(void) {
 }
 
 void *switchlink_main(void *args) {
+  pthread_mutex_lock(&rpc_start_lock);
+  while(!rpc_start_cookie) {
```

Comment:
You might want to add a space between `while` and `(`. It's a control statement, not a function call.